### ISSUE TITLE
Add `Closeable` trait

### ISSUE THREAD

**Issue by magnus-madsen** [flix/flix#8580]
Like:

```scala
import java.io.FileWriter

trait Closeable[t] {
    type Aef: Eff
    pub def close(t: t): Unit \ Closeable.Aef[t]
}

instance Closeable[FileWriter] {
    type Aef = IO
    pub def close(t: FileWriter): Unit \ IO = t.close()
}
```

Instances we should consider: FileReader, FileWriter, BufferedReader, BufferedWriter, etc. etc.

**Comment by JonathanStarup** [comment]
Doesn't this need default instances to be useful? or is this just a syntactic kind of type class like coerce?

**Comment by JonathanStarup** [comment]
Yeah but when would you abstract over it? I guess you can make a function that closes a list of closable. or even better have an instance for lists of closable things

**Comment by magnus-madsen** [comment]
> Yeah but when would you abstract over it? I guess you can make a function that closes a list of closable. or even better have an instance for lists of closable things

I see. Its like iterable. We use it for foreach. I need closeable for some scope or letregion thing.

**Comment by magnus-madsen** [comment]
We will explore this offline.
